test/parallel/test-module-loading-globalpaths.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs');
+const child_process = require('child_process');
+const pkgName = 'foo';
+
+if (process.argv[2] === 'child') {
+  console.log(require(pkgName).string);
+} else {
+  common.refreshTmpDir();
+
+  // Copy node binary into a test $PREFIX directory.
+  const prefixPath = path.join(common.tmpDir, 'install');
+  fs.mkdirSync(prefixPath);
+  let testExecPath;
+  if (common.isWindows) {
+    testExecPath = path.join(prefixPath, path.basename(process.execPath));
+  } else {
+    const prefixBinPath = path.join(prefixPath, 'bin');
+    fs.mkdirSync(prefixBinPath);
+    testExecPath = path.join(prefixBinPath, path.basename(process.execPath));
+  }
+  const mode = fs.statSync(process.execPath).mode;
+  fs.writeFileSync(testExecPath, fs.readFileSync(process.execPath));
+  fs.chmodSync(testExecPath, mode);
+
+  const runTest = (expectedString, env) => {
+    const child = child_process.execFileSync(testExecPath,
+                                             [ __filename, 'child' ],
+                                             { encoding: 'utf8', env: env });
+    assert.strictEqual(child.trim(), expectedString);
+  };
+
+  const testFixturesDir = path.join(common.fixturesDir,
+                                    path.basename(__filename, '.js'));
+
+  const env = Object.assign({}, process.env);
+  // Turn on module debug to aid diagnosing failures.
+  env['NODE_DEBUG'] = 'module';
+  // Unset NODE_PATH.
+  delete env['NODE_PATH'];
+
+  // Test empty global path.
+  const noPkgHomeDir = path.join(common.tmpDir, 'home-no-pkg');
+  fs.mkdirSync(noPkgHomeDir);
+  env['HOME'] = env['USERPROFILE'] = noPkgHomeDir;
+  assert.throws(
+    () => {
+      child_process.execFileSync(testExecPath, [ __filename, 'child' ],
+                                 { encoding: 'utf8', env: env });
+    },
+    new RegExp('Cannot find module \'' + pkgName + '\'')
```

Comment:
Done. I've been running `make lint` before committing so it disappointing the indentation isn't being flagged.